core/CronArchive.php

```diff
@@ -938,6 +934,77 @@ public function isThereExistingValidPeriod(Parameters $params)
         return !empty($idArchive);
     }
 
+    // public for tests
+    public function repairInvalidationsIfNeeded($archiveToProcess)
+    {
+        $table = Common::prefixTable('archive_invalidations');
+
+        $bind = [
+            $archiveToProcess['idsite'],
+            $archiveToProcess['name'],
+            $archiveToProcess['period'],
+            $archiveToProcess['date1'],
+            $archiveToProcess['date2'],
+        ];
+
+        $reportClause = '';
+        if (!empty($archiveToProcess['report'])) {
+            $reportClause = " AND report = ?";
+            $bind[] = $archiveToProcess['report'];
+        }
+
+        $sql = "SELECT DISTINCT period FROM `$table` WHERE idsite = ? AND name = ? AND period > ? AND ? >= date1 AND date2 >= ? $reportClause";
```

Comment:
wondering. Do we here also need to include `and status = 0`?

Reply:
👍